### PR TITLE
Refactor automatic scaling implementation for NonlinearEigenSystem

### LINKED ISSUE #15048: Automatic scaling for the new eigenvalue executioner
## Reason
Automatic scaling works for steady-state, transient and eigenvalue with our old eigenvalue executioner but does not work for eigenvalue with our new eigen executioner.

## Design
Possibly we can enable this fairly quickly by moving some of the codes in `NonlinearSystem` into `NonlinearSystemBase`, which the `NonlinearEigenSystem` the new eigen executioner works with is derived from.

## Impact
New capability.

### PR DESCRIPTION
**Pull request by lindsayad**
Moved `computeScaling` from `NonlinearSystem` down into
`NonlinearSystemBase` so that all derived classes may make use of the
method.

Closes #15048

### CODE REVIEW

File: test/tests/problems/eigen_problem/eigensolvers/ne-coupled-resid-scaling.i [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    type = CoefReaction
+    variable = u
+    coefficient = -1.0
+    extra_vector_tags = 'eigen'
+  [../]
+
+  [./diff_T]
+    type = CoefDiffusion
+    variable = T
+    coef = 1e30
```
Comment (fdkong) [inline test/tests/problems/eigen_problem/eigensolvers/ne-coupled-resid-scaling.i:44]: Oh, beautiful number. How about 1e3000?

Comment (lindsayad) [reply test/tests/problems/eigen_problem/eigensolvers/ne-coupled-resid-scaling.i:44]: Maybe we can overflow the input file

**Review by fdkong — APPROVED** [review]

**Comment by fdkong** [comment]
It looks nice to me.

**Comment by fdkong** [comment]
@YaqiWang Do you have comments?

**Comment by YaqiWang** [comment]
Sorry, did not notice this until I check recently merged PRs in MOOSE. Should be good. Glad to see you guys working on these issues!

**Pull request merged**